examples/pytorch/image-classification/run_image_classification_no_trainer.py

```diff
@@ -385,7 +391,7 @@ def collate_fn(examples):
         name=args.lr_scheduler_type,
         optimizer=optimizer,
         num_warmup_steps=args.num_warmup_steps,
-        num_training_steps=args.max_train_steps,
+        num_training_steps=math.ceil(len(train_dataloader)) * args.num_train_epochs,
```

Comment:
```suggestion
        num_training_steps=len(train_dataloader) * args.num_train_epochs,
```
This length should be an integer.

Also we are ignoring the `max_steps` argument set by the user here, I don't think this change should be made.

Reply:
Yea, I see. 

So, now i've added that its `len(train_dataloader) * args.num_train_epochs` and `args.max_train_steps * args.gradient_accumulation_steps` in case user provided max_train_steps.